pkg/pool-stable-phantom/test/StablePhantomPool.test.ts

```diff
@@ -1057,237 +1060,136 @@ describe('StablePhantomPool', () => {
-            const previousDueFee = await pool.getDueProtocolFeeBptAmount();
-
-            const tokenIn = tokens.first;
-            const tokenOut = tokens.second;
-            const { receipt } = await pool.swapGivenIn({ in: tokenIn, out: tokenOut, amount, from: lp, recipient });
-
-            const currentDueFee = await pool.getDueProtocolFeeBptAmount();
-            const aproxFee = getAproxDueFee(amount, AmountKind.WITH_FEE);
+          function getExpectedProtocolFee(amount: BigNumber, kind: AmountKind, recoveryMode: boolean): BigNumber {
+            // In StablePools, BPT and underlying tokens are almost equivalent. This means that the token fee amount is a
+            // good estimate of the equivalent BPT fee amount.
 
-            expect(currentDueFee).to.be.equalWithError(aproxFee, 0.01);
+            if (recoveryMode) {
+              return bn(0);
+            }
 
-            expectEvent.inIndirectReceipt(receipt, pool.instance.interface, 'DueProtocolFeeIncreased', {
-              bptAmount: currentDueFee.sub(previousDueFee),
-            });
-          });
+            if (kind == AmountKind.WITHOUT_FEE) {
+              amount = amount.mul(fp(1)).div(fp(1).sub(swapFeePercentage));
+            }
 
-          it('tracks fees when swapping for BPT (join)', async () => {
-            const previousDueFee = await pool.getDueProtocolFeeBptAmount();
+            const fee = amount.mul(swapFeePercentage).div(fp(1));
+            const protocolFee = fee.mul(protocolFeePercentage).div(fp(1));
+            return protocolFee;
+          }
 
-            const token = tokens.first;
-            const { amountOut: bptAmount, receipt } = await pool.swapGivenIn({
-              in: token,
-              out: pool.bpt,
-              amount,
-              from: lp,
-              recipient,
-            });
+          context('on swaps given in', () => {
+            it('pays any protocol fees due when swapping tokens', async () => {
+              const previousBalance = await pool.balanceOf(protocolFeesCollector.address);
 
-            const currentDueFee = await pool.getDueProtocolFeeBptAmount();
-            const aproxFee = getAproxDueFee(bptAmount, AmountKind.WITHOUT_FEE);
+              const tokenIn = tokens.first;
+              const tokenOut = tokens.second;
+              await pool.swapGivenIn({ in: tokenIn, out: tokenOut, amount, from: lp, recipient });
 
-            expect(currentDueFee).to.be.equalWithError(aproxFee, 0.01);
+              const currentFee = await pool.balanceOf(protocolFeesCollector.address);
+              const expectedFee = getExpectedProtocolFee(amount, AmountKind.WITH_FEE, inRecoveryMode);
 
-            expectEvent.inIndirectReceipt(receipt, pool.instance.interface, 'DueProtocolFeeIncreased', {
-              bptAmount: currentDueFee.sub(previousDueFee),
+              expect(currentFee).to.be.equalWithError(expectedFee.sub(previousBalance), 0.01);
```

Comment:
These should be `currentFee.sub(previousBalance) equal expectedFee`. I'd suggest renaming `currentFee` to `currentBalance`.

Reply:
Yes; the naming made it confusing.